examples/plot_how_train_test_and_tune.py

```diff
@@ -0,0 +1,669 @@
+
+
+def plot_simple_train_test(ax, windows_dataset, train_set, test_set):
+    """Create a sample plot for training-testing split."""
+    braindecode_cmap = ["#3A6190", "#683E00", "#DDF2FF", "#2196F3"]
+
+    ax.scatter(
+        range(len(windows_dataset)),
+        [3.5] * len(windows_dataset),
+        c=braindecode_cmap[0],
+        marker="_",
+        lw=50,
+    )
+
+    ax.scatter(
+        range(len(train_set) + len(test_set)),
+        [0.5] * len(train_set) + [0.5] * len(test_set),
+        c=[braindecode_cmap[1]] * len(train_set)
+        + [braindecode_cmap[2]] * len(test_set),
+        marker="_",
+        lw=50,
+    )
+
+    ax.set(
+        ylim=[-1, 5],
+        yticks=[0.5, 3.5],
+        yticklabels=["Train-Test\nSplit", "Original\nDataset"],
+        xlabel="Number of samples.",
+        title="Train-Test Split",
+    )
+
+    ax.legend(
+        [
+            Patch(color=braindecode_cmap[0]),
+            Patch(color=braindecode_cmap[1]),
+            Patch(color=braindecode_cmap[2]),
+        ],
+        ["Original set", "Training set", "Testing set"],
+        loc=(1.02, 0.8),
+    )
+    return ax
+
+
+fig, ax = plt.subplots(figsize=(12, 5))
+plot_simple_train_test(
+    ax=ax, windows_dataset=windows_dataset, train_set=train_set, test_set=test_set
+)
+fig.tight_layout()
+######################################################################
+# Option 2: Train-Val-Test Split
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+######################################################################
+# When evaluating different settings hyperparameters for your model,
+# there is still a risk of overfitting on the test set because the
+# parameters can be tweaked until the estimator performs optimally.
+# For more information visit `sklearns Cross-Validation Guide
+# <https://scikit-learn.org/stable/modules/cross_validation.html>`__.
+# This second option splits the original ``train_set`` into two distinct
+# sets, the training set and the validation set to avoid overfitting
+# the hyperparameters to the test set.
+#
+# .. note::
+#    If your dataset is really small, the validation split can become
+#    quite small. This may lead to unreliable tuning results. To
+#    avoid this, either use Option 3 or adjust the split ratio.
+#
+# To split the ``train_set`` we will make use of the
+# ``train_split`` argument of ``EEGClassifier``. If you leave this empty
+# (not None!), skorch will make an 80-20 train-validation split.
+# If you want to control the split manually you can do that by using
+# ``Subset`` from torch and ``predefined_split`` from skorch.
+#
+
+from torch.utils.data import Subset
+from sklearn.model_selection import train_test_split
+from skorch.helper import predefined_split, SliceDataset
+
+X_train = SliceDataset(train_set, idx=0)
+y_train = np.array([y for y in SliceDataset(train_set, idx=1)])
+train_indices, val_indices = train_test_split(
+    X_train.indices_, test_size=0.2, shuffle=False
+)
```

Comment:
this is neat but a bit convoluted for an intro tutorial.

I would consider exposing in train_set indices or something that can simply
be passed to train_test_split

Reply:
One could use
```
train_test_split(np.array(len(train_set)), test_size=0.2, shuffle=False)
```
instead, but `X_train` and `y_train` would still be necessary for `cross_val_score, search.fit()` and the visualizations.